vulture_os/toolkit/api_parser/trendmicro_visionone/trendmicro_visionone.py

```diff
@@ -0,0 +1,180 @@
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+
+class TrendmicroVisionOneAPIError(Exception):
+    pass
+
+
+class TrendmicroVisiononeParser(ApiParser):
+    URL_BASE = "https://api.eu.xdr.trendmicro.com"
+
+    def __init__(self, data):
+        """
+            trendmicro_visionone_token
+
+        """
+        super().__init__(data)
+
+        self.trendmicro_visionone_token = data["trendmicro_visionone_token"]
+        self.trendmicro_visionone_alerts_timestamp = data.get("trendmicro_visionone_alerts_timestamp")
+        self.trendmicro_visionone_audit_timestamp = data.get("trendmicro_visionone_audit_timestamp")
+        self.trendmicro_visionone_oat_timestamp = data.get("trendmicro_visionone_oat_timestamp")
+
+
+    def _get_alerts(self, since=None, to=None):
+
+        url_path = '/v3.0/workbench/alerts'
+
+        query_params = {'startDateTime': since,
+                        'endDateTime': to,
+                        'dateTimeTarget': 'createdDateTime',
+                        'orderBy': 'createdDateTime desc'}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token, 'TMV1-Filter': ""}
+
+        result = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+        if result.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {result.status_code} Reason/Content: {result.content}")
+
+        return result.json()['items']
+
+    def _get_auditlogs(self, since=None, to=None):
+        url_path = '/v3.0/audit/logs'
+
+        query_params = {'startDateTime': since,
+                        'endDateTime': to,
+                        'orderBy': 'createdDateTime desc',
+                        'top': 50,
+                        'labels': 'all'}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token,
+                   'Accept': 'application/json',
+                   'TMV1-Filter': ''}
+
+        r = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
+
+        if r.status_code != 200:
+            raise TrendmicroVisionOneAPIError(
+                f"Error on URL: {self.URL_BASE + url_path} Status: {r.status_code} Reason/Content: {r.content}")
+
+        return r.json()['items']
+
+    def _get_OAT(self, since=None, to=None):
+
+        url_path = '/v3.0/audit/logs'
+
+        query_params = {'detectedStartDateTime': since,
+                        'detectedEndDateTime': to,
+                        'ingestedStartDateTime': since,
+                        'ingestedEndDateTime': to,
+                        'top': 50}
+        headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token,
+                   'Accept': 'application/json',
+                   'TMV1-Filter': ''}
+
+        r = requests.get(self.URL_BASE + url_path, params=query_params, headers=headers)
```

Comment:
missing on every request :
- use of session to avoid having to provide headers (especially, to avoid forgetting them)
- use of proxy when available
- use of explicit timeouts to avoid overly long requests